web/components/manalinks/create-links-button.tsx

```diff
@@ -63,15 +63,39 @@ function CreateManalinkForm(props: {
   const [finishedCreating, setFinishedCreating] = useState(false)
   const [copyPressed, setCopyPressed] = useState(false)
   setTimeout(() => setCopyPressed(false), 300)
+  const defaultExpire = 'week'
+  const [expiresIn, setExpiresIn] = useState(defaultExpire)
 
   const [newManalink, setNewManalink] = useState<ManalinkInfo>({
-    expiresTime: null,
+    expiresTime: dayjs().add(1, defaultExpire).valueOf(),
     amount: 100,
     maxUses: 1,
     uses: 0,
     message: '',
   })
 
+  const EXPIRE_OPTIONS = {
+    day: '1 Day',
+    week: '1 Week',
+    month: '1 Month',
+    never: 'Never',
+  }
+
+  const expireOptions = Object.entries(EXPIRE_OPTIONS).map(([key, value]) => {
+    return <option value={key}>{value}</option>
+  })
+
+  function setExpireTime(timeDelta: string) {
+    const expiresTime =
+      timeDelta === 'never' ? null : dayjs().add(1, timeDelta).valueOf()
+    setNewManalink((m) => {
+      return {
+        ...m,
+        expiresTime: expiresTime,
```

Comment:
not sure if you knew, but when the key and the value in a JS object have the same variable name like this, you can actually omit the value! aka `return {...m, expiresTime}` would just work, I think :)

Reply:
changed! Everyday I learn new things
![image](https://user-images.githubusercontent.com/46611122/179124160-83d88794-399e-4815-a748-b633f60eb0d4.png)